security/pkg/server/ca/server.go

```diff
@@ -108,10 +108,10 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 func (s *Server) HandleCSR(ctx context.Context, request *pb.CsrRequest) (*pb.CsrResponse, error) {
 	s.monitoring.CSR.Inc()
 	caller := s.authenticate(ctx)
-	if caller == nil {
-		log.Warn("request authentication failure")
+	if caller == nil || len(caller.Identities) == 0 {
```

Comment:
Can you explain why this can fix the error? It seems to me only add another  error case to fail csr. The original issue however is complaining the cert has empty SAN.

Reply:
The real fix is in line 135, where we pass the caller.Identities to ca.Sign instead of []string{} :)

Reply:
If line 135 use an empty string array, why would the k8s case can get the right certificate? Are they using different code path?

Reply:
The k8s case doesn't use this function. It's using the other function to serve the new proto:
https://github.com/istio/istio/blob/master/security/pkg/server/ca/server.go#L71
We will merge them together in the future.